### PR TITLE
no 'Follow' ghost option to simple mob emotes

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request

no 'Follow' ghost option to simple mob emotes.
chat will not show `(F)` button on emotes when they have no mind

## Why It's Good For The Game
better chat readability for ghosts

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

--------

image

--------

image

--------

image

ghost option
---------

image

--------

image

-----------

image
image

 

</details>

## Changelog
:cl:
add: ghost chat pref to toggle (F) option to all mobs or mind-having mobs in chat. with this, ghost will not see (F) button on emotes that are done by mobs having no mind. new players has this turned on by default (they see all), but old players should set this toggle if they want to see all mobs emote
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
What if I want to follow them? There is a preference to disable ghost vision, there should be a preference to disable ghost vision for mindless mobs instead (although diseases make it practically impossible to observe while on anyway)

**Comment by EvilDragonfiend** [comment]
> What if I want to follow them? There is a preference to disable ghost vision, there should be a preference to disable ghost vision for mindless mobs instead (although diseases make it practically impossible to observe while on anyway)

I guess I will just use 'Jump To Mob'

**Comment by ivanmixo** [comment]
Literally why? If we're going to have the line shown to us as a ghost we might as well have the 3 extra characters to follow.

**Comment by EvilDragonfiend** [comment]
> Literally why? If we're going to have the line shown to us as a ghost we might as well have the 3 extra characters to follow.

first, it's annoying to misclick
second, It's a bit disruptive to read other chat lines since everything says (F), and having no it would easily make you it's not important.
third, it's mindless mob. why do you really need to track it?

**Comment by ivanmixo** [comment]
Again, if we're going to print the line we might as well have the option to follow it. Just make a pref that disables emotes from mindless mobs for ghosts.

**Comment by EvilDragonfiend** [comment]
added, but you'll need to toggle it back if you want to see (F) from all mobs due to how pref works.

File: code/modules/client/preferences_toggles.dm [hunk 1]
```diff
@@ -59,6 +59,15 @@
 	prefs.save_preferences()
 	SSblackbox.record_feedback("nested tally", "preferences_verb", 1, list("Toggle Ghost Laws", "[prefs.chat_toggles & CHAT_GHOSTLAWS ? "Enabled" : "Disabled"]")) //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
 
+/client/verb/toggle_ghost_follow()
+	set name = "Toggle Ghost (F) Chat"
```
Comment (PowerfulBacon) [inline code/modules/client/preferences_toggles.dm]:
```suggestion
	set name = "Toggle NPC GhostFollow"
```

File: code/modules/client/preferences_toggles.dm [hunk 2]
```diff
@@ -59,6 +59,15 @@
...
+	set category = "Preferences"
+	set desc = "See All (F) feature in chat"
```
Comment (PowerfulBacon) [inline code/modules/client/preferences_toggles.dm]:
```suggestion
	set desc = "Toggle the visibility of the follow button for NPC messages and emotes."
```

File: code/modules/client/preferences_toggles.dm [hunk 3]
```diff
@@ -59,6 +59,15 @@
...
+	prefs.chat_toggles ^= CHAT_GHOSTFOLLOWMINDLESS
+	to_chat(usr, "As a ghost, you will now [(prefs.chat_toggles & CHAT_GHOSTFOLLOWMINDLESS) ? "see <span class='notice'>(F)</span> from all mobs" : "see <span class='notice'>(F)</span> from only mobs having a mind"].")
+	prefs.save_preferences()
+	SSblackbox.record_feedback("nested tally", "preferences_verb", 1, list("Toggle Ghost (F) Chat", "[prefs.chat_toggles & CHAT_GHOSTFOLLOWMINDLESS ? "Enabled" : "Disabled"]"))
```
Comment (PowerfulBacon) [inline code/modules/client/preferences_toggles.dm]:
```suggestion
	SSblackbox.record_feedback("nested tally", "preferences_verb", 1, list("Toggle NPC GhostFollow", "[prefs.chat_toggles & CHAT_GHOSTFOLLOWMINDLESS ? "Enabled" : "Disabled"]"))
```

File: code/modules/client/preferences_toggles.dm [hunk 4]
```diff
@@ -59,6 +59,15 @@
 	prefs.save_preferences()
 	SSblackbox.record_feedback("nested tally", "preferences_verb", 1, list("Toggle Ghost Laws", "[prefs.chat_toggles & CHAT_GHOSTLAWS ? "Enabled" : "Disabled"]")) //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
 
+/client/verb/toggle_ghost_follow()
+	set name = "Toggle Ghost (F) Chat"
+	set category = "Preferences"
+	set desc = "See All (F) feature in chat"
+	prefs.chat_toggles ^= CHAT_GHOSTFOLLOWMINDLESS
+	to_chat(usr, "As a ghost, you will now [(prefs.chat_toggles & CHAT_GHOSTFOLLOWMINDLESS) ? "see <span class='notice'>(F)</span> from all mobs" : "see <span class='notice'>(F)</span> from only mobs having a mind"].")
```
Comment (PowerfulBacon) [inline code/modules/client/preferences_toggles.dm]:
```suggestion
	to_chat(usr, "As a ghost, you will [(prefs.chat_toggles & CHAT_GHOSTFOLLOWMINDLESS) ? "now" : "no longer"] see the follow button for NPC mobs.")
```

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
I think using the term 'NPC' mobs is an easier and shorter way to describe mobs with no minds for players that do not know what a mind is in code.

**Comment by ivanmixo** [comment]
This is still super dumb... Why not make the pref just disable printing of emotes from mindless mobs for ghosts instead of it just removing 3 characters and directly downgrading the line being output? (if you don't want to be able to follow them then why see it at all?)

**Comment by EvilDragonfiend** [comment]
> This is still super dumb... Why not make the pref just disable printing of emotes from mindless mobs for ghosts instead of it just removing 3 characters and directly downgrading the line being output? (if you don't want to be able to follow them then why see it at all?)

because the main point of this PR is to solve some clicking issues, not seeing emotes from random mobs.

**Comment by ivanmixo** [comment]
Never seen a bigger non issue in my life

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**